src/libaktualizr/storage/invstorage.cc

```diff
@@ -216,7 +216,8 @@ void INvStorage::FSSToSQLS(FSStorageRead& fs_storage, SQLStorage& sql_storage) {
 }
 
 bool INvStorage::fsReadInstalledVersions(const boost::filesystem::path& filename,
-                                         std::vector<Uptane::Target>* installed_versions, size_t* current_version) {
+                                         std::vector<Uptane::Target>* installed_versions, size_t* current_version,
```

Comment:
Is it intentional that `importInstalledVersions` was not changed to adopt these changes? It's a code part that should get more hits than `FSStorageRead::loadInstalledVersions` which is only used for legacy migrations.

Reply:
Sorta. This change was a bit of a hack to get tests to work. `importInstalledVersions` should only run on a primary, and there is already logic to assume it's for the primary when reading the installation versions in the function that converts FS to SQL storage. However, the tests had to do some fakery to test that effectively, so I had to force the serial/hwid. Would you prefer that `importInstalledVersions` also used it? In that case, I probably wouldn't need the parameters to be optional; I think I could make them a requirement.

Reply:
That's a bit annoying to have this almost dead code to make the tests work indeed. Then it should maybe explained somewhere with a good comment (maybe I've missed it?), as I found it a bit confusing in this state.

Reply:
There is no comment at present. The other option is to do some more hackery in the test itself, which is maybe safer, but makes the test slightly less meaningful. It's kind of ugly no matter how you slice it, but I'm open to advice.